File: app/schemas/content.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, HttpUrl


class ContentCreate(BaseModel):
    title: str
    url: Optional[HttpUrl] = None
    raw_content: Optional[str] = None
    content_type: str = "url"
    is_public: bool = False


class ContentUpdate(BaseModel):
    title: Optional[str] = None
    is_public: Optional[bool] = None


class ContentRead(BaseModel):
    id: int
    title: str
    url: Optional[str] = None
    content_type: str
    summary: Optional[str] = None
    tags: Optional[List[str]] = None
    processing_error: Optional[str] = None
    status: str
    is_public: bool
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/content.ts
export type ContentType = "url" | "pdf" | "text";

export type ContentStatus = "pending" | "processing" | "completed" | "failed";

// FastAPI의 ContentRead 스키마에 대응
export type ContentItem = {
  id: number;
  title: string;
  url?: string | null;
  content_type: string;
  summary?: string | null;
  tags?: string[] | null;
  status: ContentStatus;
  is_public: boolean;
  created_at: string;
  updated_at?: string | null;
};

export type SearchChunk = {
  content_id: number;
  chunk_index: number;
  chunk_text: string;
  title: string;
  summary: string;
  tags: string[];
  similarity_score: number;
  user_id: number;
};

export type SearchResultItem = {
  content_id: number;
  title: string;
  summary: string;
  tags: string[];
  similarity_score: number;
  user_id: number;
  top_snippet: string;
  matched_chunks: SearchChunk[];
};

export type SemanticSearchResponse = {
  query: string;
  total: number;
  results: SearchResultItem[];
  search_type: string;
};

export type ChatSource = {
  content_id: number;
  title: string;
  chunk_index: number;
  snippet: string;
  similarity_score: number;
};

export type ChatAnswer = {
  answer: string;
  sources: ChatSource[];
  confidence: number;
};

